web/packages/teleport/src/components/FormLogin/FormLogin.tsx

```diff
@@ -48,67 +49,65 @@ import { StepSlider, StepComponentProps } from 'design/StepSlider';
 
+const allAuthTypes: PrimaryAuthType[] = ['passwordless', 'sso', 'local'];
+
 export default function LoginForm(props: Props) {
   const {
-    title,
     attempt,
     isLocalAuthEnabled = true,
+    isPasswordlessEnabled,
     authProviders = [],
+    primaryAuthType,
   } = props;
 
   const ssoEnabled = authProviders?.length > 0;
 
   // If local auth was not enabled, disregard any primary auth type config
   // and display sso providers if any.
-  if (!isLocalAuthEnabled && ssoEnabled) {
-    return (
-      <Card my="5" mx="auto" width="464px" pb={4}>
-        <Text typography="h3" pt={4} textAlign="center">
-          {title}
-        </Text>
-        {attempt.isFailed && (
-          <Alerts.Danger m={5} mb={0}>
-            {attempt.message}
-          </Alerts.Danger>
-        )}
-        <SsoList {...props} autoFocus={true} hasTransitionEnded={true} />
-      </Card>
-    );
-  }
+  const actuallyPrimaryType = isLocalAuthEnabled ? primaryAuthType : 'sso';
 
-  if (!isLocalAuthEnabled) {
-    return (
-      <Card my="5" mx="auto" width="464px" px={5} pb={4}>
-        <Text typography="h3" pt={4} textAlign="center">
-          {title}
-        </Text>
-        <Alerts.Danger my={5}>Login has not been enabled</Alerts.Danger>
-        <Text mb={2} typography="paragraph2" width="100%">
-          The ability to login has not been enabled. Please contact your system
-          administrator for more information.
-        </Text>
-      </Card>
-    );
+  const allowedAuthTypes = allAuthTypes.filter(t => {
+    if (!isLocalAuthEnabled) return ssoEnabled && t === 'sso';
+    if (!isPasswordlessEnabled && t === 'passwordless') return false;
+    if (!ssoEnabled && t === 'sso') return false;
+    return true;
+  });
+  const otherAuthTypes = allowedAuthTypes.filter(
+    t => t !== actuallyPrimaryType
+  );
+
+  let errorMessage = '';
+  if (allowedAuthTypes.length === 0) {
+    errorMessage = 'Login has not been enabled';
+  } else if (attempt.isFailed) {
+    errorMessage = attempt.message;
   }
 
   // Everything below requires local auth to be enabled.
   return (
-    <Card my="5" mx="auto" width={464} pb={4}>
-      <Text typography="h3" pt={4} textAlign="center">
-        {title}
+    <Card my="5" mx="auto" width={650} py={4}>
+      <Text typography="h3" mb={4} textAlign="center">
+        Sign in to Teleport
       </Text>
-      {attempt.isFailed && (
-        <Alerts.Danger m={5} mb={0}>
-          {attempt.message}
-        </Alerts.Danger>
+      {errorMessage && <Alerts.Danger m={4}>{errorMessage}</Alerts.Danger>}
+      {allowedAuthTypes.length > 0 ? (
+        <StepSlider<typeof loginViews>
+          flows={loginViews}
+          currFlow={'default'}
+          otherAuthTypes={otherAuthTypes}
+          {...props}
+          primaryAuthType={actuallyPrimaryType}
+        />
+      ) : (
+        <Text mx={4} typography="paragraph2">
+          The ability to login has not been enabled. Please contact your system
```

Comment:
Out of curiosity, is this actually possible?

Reply:
That's a very good question. I could have sworn I've seen this message once, but now I don't remember if it was actually in the wild, or just in the storybook.

Reply:
OK, I found the case. It is possible to configure (or, should I say, misconfigure) the system the following way:

```yaml
auth_service:
  authentication:
    type: local
    local_auth: false
```